doc/source/train/doc_code/lightgbm_quickstart.py

```diff
@@ -0,0 +1,116 @@
+# flake8: noqa
+# isort: skip_file
+
+# __lightgbm_start__
+import pandas as pd
+import lightgbm as lgb
+
+# 1. Load your data as a `lightgbm.Dataset`.
+train_df = pd.read_csv("s3://ray-example-data/iris/train/1.csv")
+eval_df = pd.read_csv("s3://ray-example-data/iris/val/1.csv")
+
+train_X = train_df.drop("target", axis=1)
+train_y = train_df["target"]
+eval_X = eval_df.drop("target", axis=1)
+eval_y = eval_df["target"]
+
+train_set = lgb.Dataset(train_X, label=train_y)
+eval_set = lgb.Dataset(eval_X, label=eval_y)
+
+# 2. Define your LightGBM model training parameters.
+params = {
+    "objective": "multiclass",
+    "num_class": 3,
+    "metric": ["multi_logloss", "multi_error"],
+    "verbosity": -1,
+    "boosting_type": "gbdt",
+    "num_leaves": 31,
+    "learning_rate": 0.05,
+    "feature_fraction": 0.9,
+    "bagging_fraction": 0.8,
+    "bagging_freq": 5,
+}
+
+# 3. Do non-distributed training.
+model = lgb.train(
+    params,
+    train_set,
+    valid_sets=[eval_set],
+    valid_names=["eval"],
+    num_boost_round=100,
+)
+# __lightgbm_end__
+
+
+# __lightgbm_ray_start__
+import lightgbm as lgb
+
+import ray.train
+from ray.train.lightgbm import LightGBMTrainer, RayTrainReportCallback
+
+# 1. Load your data as a Ray Data Dataset.
+train_dataset = ray.data.read_csv("s3://anonymous@ray-example-data/iris/train")
+eval_dataset = ray.data.read_csv("s3://anonymous@ray-example-data/iris/val")
+
+
+def train_func():
+    # 2. Load your data shard as a `lightgbm.Dataset`.
+
+    # Get dataset shards for this worker
+    train_shard = ray.train.get_dataset_shard("train")
+    eval_shard = ray.train.get_dataset_shard("eval")
+
+    # Convert shards to pandas DataFrames
+    train_df = train_shard.materialize().to_pandas()
+    eval_df = eval_shard.materialize().to_pandas()
+
+    train_X = train_df.drop("target", axis=1)
+    train_y = train_df["target"]
+    eval_X = eval_df.drop("target", axis=1)
+    eval_y = eval_df["target"]
+
+    train_set = lgb.Dataset(train_X, label=train_y)
+    eval_set = lgb.Dataset(eval_X, label=eval_y)
+
+    # 3. Define your LightGBM model training parameters.
+    params = {
+        "objective": "multiclass",
+        "num_class": 3,
+        "metric": ["multi_logloss", "multi_error"],
+        "verbosity": -1,
+        "boosting_type": "gbdt",
+        "num_leaves": 31,
+        "learning_rate": 0.05,
+        "feature_fraction": 0.9,
+        "bagging_fraction": 0.8,
+        "bagging_freq": 5,
+    }
+
```

Comment:
need to add the network params:

```python
            # 2. Run distributed data-parallel training.
            # `get_network_params` sets up the necessary configurations for LightGBM
            # to set up the data parallel training worker group on your Ray cluster.
            params = {
                "objective": "regression",
                # Adding the line below is the only change needed
                # for your `lgb.train` call!
                **ray.train.lightgbm.v2.get_network_params(),
            }

```

see https://github.com/ray-project/ray/blob/655f736a5493ad965afec54eeb90ec31aeb80825/python/ray/train/lightgbm/v2.py#L47

Reply:
oops, good catch